dll/win32/shell32/dialogs/filedefext.cpp

```diff
@@ -552,7 +557,26 @@ CFileDefExt::InitFileAttr(HWND hwndDlg)
             FileSize.u.LowPart = FileInfo.nFileSizeLow;
             FileSize.u.HighPart = FileInfo.nFileSizeHigh;
             if (SH_FormatFileSizeWithBytes(&FileSize, wszBuf, _countof(wszBuf)))
+            {    
                 SetDlgItemTextW(hwndDlg, 14011, wszBuf);
+                // Calculate size on disc
+                // Calculate size on disc
+                // if disk usage is not aligned on a complete disk geometry block (sector/cluster), align on the next block size, depending on disk geometry data
+                // This could be improved as per : https://docs.microsoft.com/en-us/previous-versions/technet-magazine/hh148159(v=msdn.10)
+                if (GetVolumePathName(m_wszPath, szVolumePathName, _countof(szVolumePathName)))
+                {
+                    if (GetDiskFreeSpace(szVolumePathName, &ulSectorsPerCluster, &ulBytesPerSector, NULL, NULL))
+                    {
+                        ulBlockSize = ulBytesPerSector*ulSectorsPerCluster;
+                        if(FileSize.QuadPart % (ulBlockSize))
+                        {
+                            FileSize.QuadPart = ((FileSize.QuadPart / (ulBlockSize))+1)*ulBlockSize;
```

Comment:
```suggestion
                            FileSize.QuadPart += ulBlockSize - 1;
                            FileSize.QuadPart /= ulBlockSize;
                            FileSize.QuadPart *= ulBlockSize;
```

Reply:
For the sake of lisibility of the formula, I prefer the original, while I reformat it in a more compliant way